library/image_streamer_artifact_bundle.py

```diff
@@ -0,0 +1,335 @@
+                changed, msg, ansible_facts = self.__upload_backup(data)
+
+            elif state == 'backup_created':
+                changed, msg, ansible_facts = self.__create_backup(data)
+
+            elif state == 'extracted':
+                changed, msg, ansible_facts = self.__extract(resource)
+            elif state == 'backup_extracted':
+                changed, msg, ansible_facts = self.__extract_backup(data)
+
+            self.module.exit_json(changed=changed, ansible_facts=ansible_facts)
+
+        except Exception as exception:
+            self.module.fail_json(msg='; '.join(str(e) for e in exception.args))
+
+    def __get_by_name(self, name):
+        if name is None:
+            return None
+        return (self.i3s_client.artifact_bundles.get_by('name', name) or [None])[0]
+
+    def __present(self, data, resource):
+        if data.get('newName'):
+            changed, msg, resource = self.__update(data, resource)
+        elif data.get('localArtifactBundleFilePath'):
+            changed, msg, resource = self.__upload(data)
+        elif not resource:
+            changed, msg, resource = self.__create(data)
+        else:
+            changed = False
+            msg = ARTIFACT_BUNDLE_ALREADY_EXIST
+        return changed, msg, resource
+
+    def __absent(self, resource):
+        if resource:
+            self.i3s_client.artifact_bundles.delete(resource)
+            changed = True
+            msg = ARTIFACT_BUNDLE_DELETED
+        else:
+            changed = False
+            msg = ARTIFACT_BUNDLE_ABSENT
+        return changed, msg, {}
+
+    def __create(self, data):
+        data['buildPlans'] = [data['buildPlans']]
+        resource = self.i3s_client.artifact_bundles.create(data)
+        return True, ARTIFACT_BUNDLE_CREATED, resource
+
+    def __update(self, data, resource):
+        if resource is None:
+            resource = self.__get_by_name(data['newName'])
+        data["name"] = data.pop("newName")
+        merged_data = resource.copy()
+        merged_data.update(data)
+        if not resource_compare(resource, merged_data):
+            resource = self.i3s_client.artifact_bundles.update(merged_data)
+            changed = True
+            msg = ARTIFACT_BUNDLE_UPDATED
+        else:
+            changed = False
+            msg = ARTIFACT_BUNDLE_ALREADY_EXIST
+        return changed, msg, resource
+
+    def __download(self, data, resource):
+        self.i3s_client.artifact_bundles.download_artifact_bundle(resource['uri'], data['destinationFilePath'])
+        return False, ARTIFACT_BUNDLE_DOWNLOADED, {}
+
+    def __download_archive(self, data, resource):
+        self.i3s_client.artifact_bundles.download_archive_artifact_bundle(resource['uri'], data['destinationFilePath'])
+        return False, ARCHIVE_DOWNLOADED, {}
+
+    def __upload(self, data):
+        file_name = data['localArtifactBundleFilePath']
+        file_name = file_name.split('/')[-1].split('.')[0]
```

Comment:
In Unix systems, dots are not so common in filenames. It will crash.
And we can also enter a filename without the path, can't we?
I suggest you to use one of these functions:
[https://docs.python.org/2/library/os.path.html](https://docs.python.org/2/library/os.path.html)